src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -386,10 +386,12 @@ public BulkLoadRegisterSDFResponseDTO registerSdf(BulkLoadRegisterSDFRequestDTO
 				// Parent
 				Parent parent;
 				SaltForm saltForm;
+				boolean saltStripped = false;
 				Lot lot;
 				// attempt to strip salts
 				try {
 					mol = processForSaltStripping(mol, mappings, results, numRecordsRead);
+					saltStripped = true;
```

Comment:
I don't think this actually checks whether a salt was extracted. that `processForSaltStripping` will run on every molecule, and will just return the same `mol` if there were no salts found.
I think you need to check the properties of the `mol` before & after Salt Stripping to see if any salt info was present and/or if any salt info was added.